MapleServer2/PacketHandlers/Login/CharacterManagementHandler.cs

```diff
@@ -89,46 +86,101 @@ public void HandleCreate(LoginSession session, PacketReader packet)
                 switch (type)
                 {
                     case ItemSlot.HR: // Hair
-                        packet.Skip(56); // Hair Position
-                        Equips.Add(ItemSlot.HR, new Item(Convert.ToInt32(id)));
+                        // Hair Length/Position
+                        float backLength = BitConverter.ToSingle(packet.Read(4), 0);
+                        byte[] backPositionArray = packet.Read(24);
+                        float frontLength = BitConverter.ToSingle(packet.Read(4), 0);
+                        byte[] frontPositionArray = packet.Read(24);
+
+                        Equips.Add(ItemSlot.HR, new Item(Convert.ToInt32(id)) {
+                            Uid = BitConverter.ToInt64(Guid.NewGuid().ToByteArray(), 0),
+                            CreationTime = 1565575851,
```

Comment:
We probably don't want to hardcode the creation time. I recommend getting the current DateTime outside of the for loop and converting it to epoch time (that's what the hardcoded value is). There is a built in method that can allow you to do this conversion within the .NET framework.

https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimemilliseconds?view=netframework-4.8&viewFallbackFrom=netframework-4.0

You can easily double check and see if the value you're returning makes sense, there are multiple converters online, such as this https://www.epochconverter.com/